raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -477,6 +492,304 @@ function makeAndSignWithdrawConfirmation(
+          state.received[secrethash].transfer[1].message_identifier.eq(msgId)
+        ) {
+          // transferProcessed again will trigger messageSend.request
+          return of(
+            transferProcessed({ message: state.received[secrethash].transferProcessed![1] }, meta),
+          );
+        } else return EMPTY;
+      }
+
+      // full balance proof validation
+      const tokenNetwork = transfer.token_network_address;
+      const partner = action.meta.address;
+      const channel = state.channels[tokenNetwork]?.[partner];
+      assert(channel?.state === ChannelState.open, 'channel not found or not open');
+      assert(transfer.chain_id.eq(network.chainId), 'chainId mismatch');
+      assert(transfer.channel_identifier.eq(channel.id), 'channelId mismatch');
+      assert(transfer.nonce.eq(nextNonce(channel.partner.balanceProof)), 'nonce mismatch');
+      assert(
+        transfer.transferred_amount.eq(channel.partner.balanceProof?.transferredAmount ?? Zero),
+        'transferredAmount mismatch',
+      );
+      assert(
+        transfer.locked_amount.eq(
+          (channel.partner.balanceProof?.lockedAmount ?? Zero).add(transfer.lock.amount),
+        ),
+        'lockedAmount mismatch',
+      );
+
+      const partnerCapacity = channelAmounts(channel).partnerCapacity;
+      assert(
+        transfer.lock.amount.lte(partnerCapacity),
+        'balanceProof total amount bigger than capacity',
+      );
+
+      assert(transfer.recipient === address, "Received transfer isn't for us");
+
+      assert(
+        transfer.lock.expiration.sub(state.blockNumber).gt(revealTimeout),
+        'lock expires too soon',
+      );
+      const locks = [...(channel.partner.locks ?? []), transfer.lock];
+      const locksroot = getLocksroot(locks);
+      assert(transfer.locksroot === locksroot, 'locksroot mismatch');
+      const token = findKey(state.tokens, tn => tn === tokenNetwork)! as Address;
+
+      log.info(
+        'Receiving transfer of value',
+        transfer.lock.amount.toString(),
+        'of token',
+        token,
+        ', from',
+        transfer.initiator,
+        ', through partner',
+        partner,
+      );
+
+      let request$: Observable<Signed<SecretRequest> | undefined> = of(undefined);
+      if (!caps?.[Capabilities.NO_RECEIVE] && transfer.target === address)
+        request$ = defer(() => {
+          const request: SecretRequest = {
+            type: MessageType.SECRET_REQUEST,
+            payment_identifier: transfer.payment_identifier,
+            secrethash,
+            amount: transfer.lock.amount,
+            expiration: transfer.lock.expiration,
+            message_identifier: makeMessageId(),
+          };
+          return signMessage(signer, request, { log });
+        });
+
+      const processed$ = defer(() => {
+        const processed: Processed = {
+          type: MessageType.PROCESSED,
+          // eslint-disable-next-line @typescript-eslint/camelcase
```

Comment:
Comment is probably unnecessary, as linter does not complain about it in https://github.com/raiden-network/light-client/pull/1221/files#diff-da17a2bff4a32e6c562b937e7d182ebfR583 too?